File: app/src/main/java/com/example/phl/data/spasticity/SpasticityDao.java
package com.example.phl.data.spasticity;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import com.example.phl.data.spasticity.Spasticity;

import java.util.List;

@Dao
public interface SpasticityDao {

    @Insert
    long insert(Spasticity spasticity);

    @Query("SELECT * FROM spasticity")
    List<Spasticity> getAllSpasticities();

    @Query("SELECT * FROM spasticity WHERE id = :id")
    Spasticity getSpasticityById(int id);

    @Query("UPDATE spasticity SET value = :value WHERE id = :id")
    void updateSpasticity(int id, Double value);

    @Query("DELETE FROM spasticity WHERE id = :id")
    void deleteSpasticity(int id);
}

File: AndroidApp/app/src/main/java/com/example/phl/data/spasticity/SpasticityDao.kt
package com.example.phl.data.spasticity

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.example.phl.data.AbstractDao
import java.time.LocalDate

@Dao
interface SpasticityDao : AbstractDao<Spasticity> {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    override fun insert(data: Spasticity)

    @Query("SELECT * FROM spasticity")
    override fun getAll(): List<Spasticity>

    @Query("SELECT * FROM spasticity where sessionId = :sessionId")
    override fun getBySessionId(sessionId: String): List<Spasticity>

    @Query("SELECT * FROM spasticity WHERE sessionId = :id")
    override fun getById(id: String): Spasticity

    @Query("SELECT * FROM spasticity WHERE strftime('%Y-%m-%d', time) = :day")
    override fun getByDay(day: LocalDate): List<Spasticity>

    @Query("DELETE FROM spasticity WHERE sessionId = :id")
    override fun deleteById(id: String)

    @Query("DELETE FROM spasticity WHERE sessionId = :sessionId")
    override fun deleteBySessionId(sessionId: String)
}